src/natcap/invest/ui/model.py

```diff
@@ -1908,6 +1900,15 @@ def _quickrun_close_model():
         self.raise_()  # raise window to top of stack.
         self.validate(block=False)  # initial validation for the model
 
+        for input_obj in self.inputs:
+            input_obj.value_changed.connect(_validate)
+            try:
+                input_obj.validity_changed.connect(_validate)
+            except AttributeError:
+                # Not all inputs can have validity (e.g. Container, dropdown)
+                pass
```

Comment:
Ah, I see the issue here that this change corrects.  So under the previous setup, calling `self.load_lastrun()` would trigger the validation as each input's value was being updated, and once again when `self.validate(block=False)` is called.

Under this proposed change, is validation still being called on this set of inputs?

Offhand, it clearly makes sense to populate the inputs first and _then_ validate everything.  But it also looks like the `self.validate(block=False)` should come after this `for input_obj in self.inputs:` loop.  Does that sound right?

Reply:
`self.load_lastrun()` is called before `self.validate(block=False)`, so it is validating the autofilled inputs. I'm not sure if it matters whether the `for input_obj in self.inputs:` happens before or after `self.validate(block=False)`, so long as it's after `self.load_lastrun()`?

Reply:
> I'm not sure if it matters whether the `for input_obj in self.inputs:` block happens before or after `self.validate(block=False)`, so long as it's after `self.load_lastrun()`?

Ah, I see what's happening here, and you're right: it doesn't matter whether the `self.validate()` call is before or after the block where we connect all of the inputs' `value_changed` signals to the model windows' validation method for this use case.

My main concern was that whole-model validation was being called on the new set of inputs, and it is!  We're calling it here in this method, and the UI calls it again when the user clicks the 'Run' button in the UI.  So we should be all set here.

@emlys since the speedup on model UI launch is a user-facing change, should there be a note in `HISTORY`?

Reply:
Good point! Just added the note.